src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/FileBasedPrograms/RunApiModels.cs

```diff
@@ -54,17 +54,34 @@ internal sealed class SimpleDiagnostic
         public readonly struct Position
         {
             public string Path { get; init; }
-            public LinePositionSpan Span { get; init; }
+            public LinePositionSpanInternal Span { get; init; }
+        }
+    }
+
+    internal struct LinePositionInternal
+    {
+        public int Line { get; set; }
+        public int Character { get; set; }
+    }
```

Comment:
does this need to be mutable?

Reply:
deserialization should work if we use init accessors instead.